### ISSUE TITLE
Headlights will be on for random vehicles but not illuminate until you enter reality bubble. This causes death.

### ISSUE THREAD

**Issue by StorytimeStorey** [CleverRaven/Cataclysm-DDA#65460]
### Describe the bug

Headlights will sometimes be on for random vehicles as you explore. While a neat feature, its implementation is limited due to the reality bubble. The area isn't illuminated until you enter the reality bubble, meaning if you are hunting at night you have a chance of suddenly being illuminated while surrounded by enemies. 

### Attach save file

[Ahloso-trimmed.tar.gz](https://github.com/CleverRaven/Cataclysm-DDA/files/11391541/Ahloso-trimmed.tar.gz)

### Steps to reproduce

Attached is a save where, just to the left of the character, you can see the area suddenly become illuminated. If you go north a bit near a couple electronics stores there is another event of this except it happens while you are inside the range of the light. 

### Expected behavior

1. Have the light's illumination always be seen. I've seen some lights on from a high distance away, so I don't know why these in particular only activate their illumination when you're within such a close distance. 
or
2. Completely remove the "headlights sometimes being on" feature. It seems like it interacts strangely with reality bubbles and might just not be a good fit for how the game is structured. 

### Screenshots

![image](https://user-images.githubusercontent.com/37844773/236060980-0857c3f9-422f-4939-9ae5-018b5d9b9d49.png)
![image](https://user-images.githubusercontent.com/37844773/236060993-9b07ce45-8881-4835-a89d-cfcdaf3ee8ed.png)
![image](https://user-images.githubusercontent.com/37844773/236061004-ecf7df79-1bdb-4ef2-9394-ae7c3aa0c5fb.png)

### Versions and configuration

Windows 10
621d53e (tiles)
English
speedydex, statsthroughskills, statsthroughskills

**Comment by anoobindisguise** [comment]
Yeah it raises some interesting questions of how those lights were on since the start of the cataclysm when normally they burn out a car battery in a few hours. Same thing for \. tanks that spontaneously start draining fuel only the moment you enter their presence and were watertight until that point.

**Comment by MochaModifier** [comment]
> Yeah it raises some interesting questions of how those lights were on since the start of the cataclysm when normally they burn out a car battery in a few hours. Same thing for |. tanks that spontaneously start draining fuel only the moment you enter their presence and were watertight until that point.

sort of makes sense for solar vehicles, where the light might drain slower than the cells can charge/maintain the batteries

For other cars, it might be some story telling opportunity, where a survivor recently tried to drive through and died/fled - but no clue how to explain away the magically unsealing gas tanks. Bubble gum in the tank deciding to fall out?

**Comment by Zireael07** [comment]
> Bubble gum in the tank deciding to fall out?

LOL

**Comment by TealcOneill** [comment]
Confirmed on 195b437.

**Comment by stubkan** [comment]
I think cars being on sometimes is an excellent feature and should not be patched out.  It adds variety and a lived-in-ness to the world when you enter a city and off in the distance you can see flashing lights or a car alarm going off.

The issue this is claiming to cause isn't too much of a problem because the max range of a normal car headlight that is on will load in before the player is in range of it.  If it is turning on when a player is near - that seems to me it is just the car alarm function which occurs when a zombie bumps into the car.

**Comment by StorytimeStorey** [comment]
> The issue this is claiming to cause isn't too much of a problem because the max range of a normal car headlight that is on will load in before the player is in range of it. If it is turning on when a player is near - that seems to me it is just the car alarm function which occurs when a zombie bumps into the car.

Except when it isn't out of range when you walk into the reality bubble. I do like the feature quite a bit, but it has an edge-case issue in the programming that can kill you.

**Comment by stubkan** [comment]
> Except when it isn't out of range when you walk into the reality bubble.

No, thats not right, when you drive at night, you can see the far edge of your headlights, and it is dark beyond that.  That clearly shows that the reality bubble is larger than any headlight.

>If it is turning on when a player is near

This is most likely what you are thinking of, ie a zombie bumps into a car - or something else is setting off a car alarm, and THAT causes headlights to turn on when you are near them.  This, IMO is a good feature.

**Comment by StorytimeStorey** [comment]
> No, thats not right, when you drive at night, you can see the far edge of your headlights, and it is dark beyond that. That clearly shows that the reality bubble is larger than any headlight.

Because the light is sourced from yourself. Go to my save and go up to near an electronics store, you'll see where you enter a reality bubble with a car nearby. The car is near the edge of the reality bubble, so when you move one square right suddenly the car is "alive" and the light turns on. 


> This is most likely what you are thinking of, ie a zombie bumps into a car - or something else is setting off a car alarm, and THAT causes headlights to turn on when you are near them. This, IMO is a good feature.

There was no alarm, and going back and forth, left to right, triggered the reality bubble to turn the light off and on

**Comment by stubkan** [comment]
Your save is damaged, I cannot open it

**Comment by SurFlurer** [comment]
I noticed the same thing. The illuminated range is also weird.

![image](https://github.com/CleverRaven/Cataclysm-DDA/assets/22912139/372341df-4a76-487d-9b0a-9b86646f497d)


My savefile: 
[camptest-trimmed.tar.gz](https://github.com/CleverRaven/Cataclysm-DDA/files/11719892/camptest-trimmed.tar.gz)

**Comment by stubkan** [comment]
This screenshot illuminates why I dont see how this is a real problem - the car headlight has a range of 60 tiles, but the overmap reality bubble is 2.5 OM tiles radius, which is double 2.5 times by 24 blocks - ie 120 tiles.  

So the car will be loaded in when it enters the reality bubble about 120-100 tiles away, and then its headlight pops on and shines 60 tiles toward you, leaving a good 40-60 tiles of black space and NOT illuminating you immediately.

**Comment by SurFlurer** [comment]
> the overmap reality bubble is 2.5 OM tiles radius, which is double 2.5 times by 24 blocks - ie 120 tiles.

That's mostly true. 

> So the car will be loaded in when it enters the reality bubble about 120-100 tiles away, and then its headlight pops on and shines 60 tiles toward you, leaving a good 40-60 tiles of black space and NOT illuminating you immediately.

That's not true. You know it's radius. The reality bubble is centered on YOU, so anything further than 60 tiles from you won't be loaded. Let's use an analogy, assuming migo has 60 tiles of sight range, and they are faster than you. So the death penalty is pronounced if you see them ( as they will see you! ).

This case is similar. The car will be loaded when you are within the range of headlight.

**Comment by SurFlurer** [comment]
There's also weird icons in security camera view:
![image](https://github.com/CleverRaven/Cataclysm-DDA/assets/22912139/22b5227c-6c3a-4c3d-8b4b-a8f25c86ce41)
